pkg/apis/kubermatic/v1/cluster.go

```diff
@@ -893,11 +898,18 @@ type AWSCloudSpec struct {
 	AssumeRoleExternalID string `json:"assumeRoleExternalID,omitempty"`
 	VPCID                string `json:"vpcID"`
 	// The IAM role, the control plane will use. The control plane will perform an assume-role
-	ControlPlaneRoleARN     string `json:"roleARN"` //nolint:tagliatelle
-	RouteTableID            string `json:"routeTableID"`
-	InstanceProfileName     string `json:"instanceProfileName"`
-	SecurityGroupID         string `json:"securityGroupID"`
+	ControlPlaneRoleARN string `json:"roleARN"` //nolint:tagliatelle
+	RouteTableID        string `json:"routeTableID"`
+	InstanceProfileName string `json:"instanceProfileName"`
+	SecurityGroupID     string `json:"securityGroupID"`
+	// A CIDR range that will be used to allow access to the node port range in the security group to. Only applies if
+	// the security group is generated by KKP and not preexisting.
+	// If NodePortsAllowedIPRange nor NodePortsAllowedIPRanges is set, the node port range can be accessed from anywhere.
 	NodePortsAllowedIPRange string `json:"nodePortsAllowedIPRange,omitempty"`
+	// Optional: CIDR ranges that will be used to allow access to the node port range in the security group to. Only applies if
+	// the security group is generated by KKP and not preexisting.
+	// If NodePortsAllowedIPRange nor NodePortsAllowedIPRanges is set,  the node port range can be accessed from anywhere.
+	NodePortsAllowedIPRanges *NetworkRanges `json:"nodePortsAllowedIPRanges,omitempty"`
```

Comment:
@rastislavs is it possible to have some validation for this? Some `+kubebuilder:validation` marker, like https://github.com/ahmedwaleedmalik/kubermatic/blob/master/pkg/apis/kubermatic/v1/datacenter.go#L30

Reply:
You mean for the `NetworkRanges` type that I am reusing here?
I guess that would have to be something complex as:
```
// +kubebuilder:validation:Pattern="((^((([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]))/([0-9]|[1-2][0-9]|3[0-2])$)|(^(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:))/([0-9]|[0-9][0-9]|1[0-1][0-9]|12[0-8])$))"
```

But I am not sure if that is doable without modifying the `NetworkRanges` type - which I guess would break backward compatibility?

Reply:
> But I am not sure if that is done without modifying the NetworkRanges type - which I guess would break backward compatibility

Yes, that is correct. Maybe a different type for this case to not break backward compatibility? I know it's code duplication but at least won't break anything. But let's consider this as a "nit" and I'll leave it up to you if you want to add this validation.

Reply:
If it was just the data type, I would have no problems with duplicating it. But as you can see, `NetworkRanges` have some util methods, that would need to be duplicated as well. I would like to avoid duplicating those.

In any case, you reminded me that I forgot to add validation in the validation webhook. I'll do that. Then, if the change in the `NetworkRanges` type is not allowed, I would rely only on webhook validation.

Reply:
> But as you can see, NetworkRanges have some util methods, that would need to be duplicated as well. I would like to avoid duplicating those.

Yes, that's the main reason why I'm divided on this one because it would overcomplicate a lot of things. 

Let's just settle with the webhook validation please.